lib/ansible/modules/cloud/vmware/vmware_host_tcpip_stacks.py

```diff
@@ -0,0 +1,473 @@
+                         instance,
+                         'defaultTcpipStack',
+                         default_vmkernel_gateway,
+                         default_num_connections,
+                         default_congestion_algorithm)
+                    if changed_instance:
+                        changed = True
+                        changed_list.append("default TCP/IP stack")
+                        # vmkernel_gateway_previous can be Null
+                        if default_vmkernel_gateway:
+                            results['result'][host.name]['default_vmkernel_gateway_previous'] = vmkernel_gateway_previous
+                        if num_connections_previous:
+                            results['result'][host.name]['default_num_connections_previous'] = num_connections_previous
+                        if congestion_algorithm_previous:
+                            results['result'][host.name]['default_congestion_algorithm_previous'] = \
+                                congestion_algorithm_previous
+                        changed_instances_list.append(netstack_spec)
+                # Check Provisioning TCP/IP stack
+                if instance.key == 'vSphereProvisioning':
+                    results['result'][host.name]['prov_vmkernel_gateway'] = prov_vmkernel_gateway
+                    results['result'][host.name]['prov_vmkernel_num_connections'] = prov_vmkernel_num_connections
+                    results['result'][host.name]['prov_congestion_algorithm'] = prov_congestion_algorithm
+                    (netstack_spec,
+                     changed_instance,
+                     vmkernel_gateway_previous,
+                     num_connections_previous,
+                     congestion_algorithm_previous) = self.create_netstack_spec(
+                         instance,
+                         'vSphereProvisioning',
+                         prov_vmkernel_gateway,
+                         prov_vmkernel_num_connections,
+                         prov_congestion_algorithm)
+                    if changed_instance:
+                        changed = True
+                        changed_list.append("provisioning TCP/IP stack")
+                        # vmkernel_gateway_previous can be Null
+                        if prov_vmkernel_gateway:
+                            results['result'][host.name]['prov_vmkernel_gateway_previous'] = vmkernel_gateway_previous
+                        if num_connections_previous:
+                            results['result'][host.name]['prov_vmkernel_num_connections_previous'] = num_connections_previous
+                        if congestion_algorithm_previous:
+                            results['result'][host.name]['prov_congestion_algorithm_previous'] = \
+                                congestion_algorithm_previous
+                        changed_instances_list.append(netstack_spec)
+                # Check vMotion TCP/IP stack
+                if instance.key == 'vmotion':
```

Comment:
Look up "Rule of Three" and "Don't Repeat Yourself" to improve the code duplication situation here.

Reply:
Thanks for teaching. Do you think I'm dumb and I don't know?

Reply:
Maybe you should create some value (fix issues, create new modules, etc. pp.) instead of complaining all the time..

Reply:
@ckotte, while i understand it can be frustrating to have your work reviewed, the feedback given is always in the best interest to improve the code quality. Sometimes it might come across as aggressive, but i hope that is never the intention of the reviewer (it definitely is not for me, if it does, i apologise).